### PR TITLE
Upgrade dependencies

### LINKED ISSUE #125: Upgrade kubelet to 0.7
Upgrade kubelet to version 0.7. This includes also an upgrade of tokio to version 1.

### PR DESCRIPTION
**Pull request by siegfriedweber**
Closes #125 

# k8s-openapi
Previous version: 0.9
New version: 0.11
Newest version: 0.11

No notable changes.

# krator
Previous version: 0.1
New version: 0.2 + Patch "Krator clean-up fixed"
Newest version: 0.2

Notable changes:
* The tokio 1.x runtime is used.

A non-critical [bug](https://github.com/deislabs/krustlet/issues/556) is open which logs a warning. A [pull request](https://github.com/deislabs/krustlet/pull/557) for krator is created and the fix is also added to the Stackable fork.

# kube
Previous version: 0.42
New version: 0.48
Newest version: 0.51

kubelet 0.7 depends on kube 0.48, therefore this version was chosen.

Notable changes:
* Upgrade to tokio 1.0 and reqwest 0.11.0 required

# kube-derive
Previous version: 0.43
New version: 0.48
Newest version: 0.51

kube and kube-derive are released together so the same version was chosen.

Notable changes:
* kube-derive types now require JsonSchema derived via schemars libray.

# kubelet
Previous version: 0.6 + Patch "Draining of pods disabled" + Patch "Pod conditions"
New version: 0.7 + Patch "Draining of pods disabled"
Newest version: 0.7

Notable changes:
* The tokio 1.x runtime is used.
* Pod conditions can be added to the status. The custom patch is no longer necessary.

# oci-distribution
Previous version: 0.5
New version: 0.6
Newest version: 0.6

Notable changes:
* The tokio 1.x runtime is used.

# phf
Previous version: 0.7.24
New version: 0.8
Newest version: 0.8

Version 0.7.24 was released on 2019-01-05. Version 0.8.0 was released on 2019-10-15. The new version is not tagged in the repository. The documentation still mentions version 0.7.24. There is no changelog. We should get rid of this dependency.

# reqwest
Previous version: 0.10
New version: 0.11
Newest version: 0.11

Notable changes:
* Update to Tokio v1.0

# rstest
Previous version: 0.6
New version: 0.7
Newest version: 0.7

Notable changes:
* New syntax for test cases

# schemars
Previous version: -
New version: 0.8
Newest version: 0.8

Required now by kube-derive. Types must be derived from JsonSchema.

# tokio
Previous version: 0.2
New version: 1.4
Newest version: 1.4

Notable changes:
* Feature flags were changed.
* `time::delay_for` was renamed to `time::sleep`
* No panic on `timeout(Duration::MAX)`
* The default flavor of the main macro is `multi_thread`.

All dependencies use tokio 1.4.0 now:
```
$ cargo tree | grep "tokio "
│   │   │   │   └── tokio v1.4.0
│   │   │   │   │   ├── tokio v1.4.0 (*)
│   │   │   │   │   │   └── tokio v1.4.0 (*)
│   │   │   │   ├── tokio v1.4.0 (*)
│   │   │   │   ├── tokio v1.4.0 (*)
│   │   │   │       └── tokio v1.4.0 (*)
│   │   │   ├── tokio v1.4.0 (*)
│   │   ├── tokio v1.4.0 (*)
│   │   ├── tokio v1.4.0 (*)
│   ├── tokio v1.4.0 (*)
│   │   ├── tokio v1.4.0 (*)
│   │       ├── tokio v1.4.0 (*)
│   │       │   ├── tokio v1.4.0 (*)
│   │       │   ├── tokio v1.4.0 (*)
│   │   ├── tokio v1.4.0 (*)
│   ├── tokio v1.4.0 (*)
│       ├── tokio v1.4.0 (*)
│       │   ├── tokio v1.4.0 (*)
├── tokio v1.4.0 (*)

```

# All other dependencies

... are already up-to-date.

### CODE REVIEW

**Review by lfrancke — DISMISSED** [review]
A single comment but not important. Looks great and the update notes are fabulous!

Can we get rid of phf easily? If so...feel free to do so. I remember it was just used for a minor thing somewhere.

File: Cargo.toml [hunk 1]
```diff
@@ -10,40 +10,41 @@ license = "Apache-2.0"
...
+kube-derive = "0.48"
+kubelet = { git="https://github.com/stackabletech/krustlet.git", rev="297fe2c29207dc0c26233b906de243ca6e17148c", default-features = true, features= ["derive", "cli"] } # version = "0.7"
+lazy_static = "1.4"
 log = "0.4"
+nix = "0.20"
+oci-distribution = { git="https://github.com/stackabletech/krustlet.git", rev="297fe2c29207dc0c26233b906de243ca6e17148c"} # version = "0.6"
+phf = { version = "0.8", features = ["macros"] }
+regex = "1.4"
+reqwest = "0.11"
+schemars = "0.8"
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply Cargo.toml:30]: - Dependency to phf removed.
- Derive feature enabled on kube and dependency to kube-derive removed.
- The schema stuff is necessary for the `RepositorySpec`.

**Comment by siegfriedweber** [comment]
- Dependency to phf removed.
- `derive` feature enabled on kube and dependency to kube-derive removed.

**Review by lfrancke — APPROVED** [review]

**Review by soenkeliebau — APPROVED** [review]
LGTM - build and ran tests, as well as tested it so that it actually downloaded a package, which got broken last time we tried upgrading tokio.

I had an issue compiling this with my version of the nightly toolchain (see https://github.com/dtolnay/anyhow/issues/148 & https://github.com/dtolnay/anyhow/issues/146) but that is by _design_ apparently.

**Pull request merged**